x-pack/plugins/observability_solution/inventory/server/routes/entities/get_latest_entities.ts

```diff
@@ -28,27 +28,35 @@ export async function getLatestEntities({
   inventoryEsClient: ObservabilityElasticsearchClient;
   sortDirection: 'asc' | 'desc';
   sortField: EntityColumnIds;
-  entityTypes?: EntityType[];
+  entityTypes?: string[];
   kuery?: string;
 }) {
   // alertsCount doesn't exist in entities index. Ignore it and sort by entity.lastSeenTimestamp by default.
   const entitiesSortField = sortField === 'alertsCount' ? ENTITY_LAST_SEEN : sortField;
 
-  const request = {
-    query: `FROM ${ENTITIES_LATEST_ALIAS}
-     | ${getEntityTypesWhereClause(entityTypes)}
-     | ${getEntityDefinitionIdWhereClause()}
-     | SORT ${entitiesSortField} ${sortDirection}
-     | LIMIT ${MAX_NUMBER_OF_ENTITIES}
-     `,
+  const from = `FROM ${ENTITIES_LATEST_ALIAS}`;
+  const where: string[] = [getBuiltinEntityDefinitionIdESQLWhereClause()];
+  const params: ScalarValue[] = [];
+
+  if (entityTypes) {
+    where.push(`WHERE ${ENTITY_TYPE} IN (${entityTypes.map(() => '?').join()})`);
+    params.push(...entityTypes.map((entityType) => entityType));
+  }
+
+  const sort = `SORT ${entitiesSortField} ${sortDirection}`;
+  const limit = `LIMIT ${MAX_NUMBER_OF_ENTITIES}`;
```

Comment:
Nice! This lib https://github.com/elastic/kibana/pull/182630 could help to make it even easier to build the queries, but it's still a draft.